tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/server/QueueQueryTest.java

```diff
@@ -155,19 +156,22 @@ public void testQueueQueryOnAutoCreatedQueueWithFQQN() throws Exception {
       SimpleString addressName = SimpleString.toSimpleString(UUID.randomUUID().toString());
       SimpleString queueName = SimpleString.toSimpleString(UUID.randomUUID().toString());
       SimpleString fqqn = addressName.concat("::").concat(queueName);
-      JMSContext c = new ActiveMQConnectionFactory("vm://0").createContext();
-      c.createProducer().send(c.createQueue(fqqn.toString()), c.createMessage());
-      QueueQueryResult queueQueryResult = server.queueQuery(fqqn);
-      assertEquals(queueName, queueQueryResult.getName());
-      assertEquals(addressName, queueQueryResult.getAddress());
-      assertEquals(1, queueQueryResult.getMessageCount());
-      queueQueryResult = server.queueQuery(queueName);
-      assertEquals(queueName, queueQueryResult.getName());
-      assertEquals(addressName, queueQueryResult.getAddress());
-      assertEquals(1, queueQueryResult.getMessageCount());
-      c.createProducer().send(c.createQueue(addressName.toString()), c.createMessage());
-      assertEquals(2, server.queueQuery(fqqn).getMessageCount());
-      assertEquals(2, server.queueQuery(queueName).getMessageCount());
+      try (JMSContext c = new ActiveMQConnectionFactory("vm://0").createContext()) {
+         c.createProducer().send(c.createQueue(fqqn.toString()), c.createMessage());
+         assertTrue(Wait.waitFor(() -> server.locateQueue(queueName).getMessageCount() == 1, 2000, 100));
+         QueueQueryResult queueQueryResult = server.queueQuery(fqqn);
+         assertEquals(queueName, queueQueryResult.getName());
+         assertEquals(addressName, queueQueryResult.getAddress());
+         assertEquals(1, queueQueryResult.getMessageCount());
+         queueQueryResult = server.queueQuery(queueName);
+         assertEquals(queueName, queueQueryResult.getName());
+         assertEquals(addressName, queueQueryResult.getAddress());
+         assertEquals(1, queueQueryResult.getMessageCount());
+         c.createProducer().send(c.createQueue(addressName.toString()), c.createMessage());
+         assertTrue(Wait.waitFor(() -> server.locateQueue(queueName).getMessageCount() == 2, 2000, 100));
```

Comment:
you could have used: Wait.assertEquals(100, server.locateQueue(queueName)::getMessageCount) ?

Reply:
That's definitely more elegant. Will send an update in a bit.

Reply:
Done.